pkg/datapath/linux/node.go

```diff
@@ -393,16 +401,16 @@ func (n *linuxNodeHandler) familyEnabled(c *cidr.CIDR) bool {
 	return (c.IP.To4() != nil && n.nodeConfig.EnableIPv4) || (c.IP.To4() == nil && n.nodeConfig.EnableIPv6)
 }
 
-func (n *linuxNodeHandler) updateOrRemoveNodeRoutes(old, new []*cidr.CIDR) {
+func (n *linuxNodeHandler) updateOrRemoveNodeRoutes(old, new []*cidr.CIDR, isLocalNode bool) {
 	addedAuxRoutes, removedAuxRoutes := cidr.DiffCIDRLists(old, new)
 	for _, prefix := range addedAuxRoutes {
 		if prefix != nil {
-			n.updateNodeRoute(prefix, n.familyEnabled(prefix))
+			n.updateNodeRoute(prefix, n.familyEnabled(prefix), isLocalNode)
 		}
 	}
 	for _, prefix := range removedAuxRoutes {
-		if rt, _ := n.lookupNodeRoute(prefix); rt != nil {
-			n.deleteNodeRoute(prefix)
+		if rt, _ := n.lookupNodeRoute(prefix, false); rt != nil {
+			n.deleteNodeRoute(prefix, isLocalNode)
```

Comment:
I don't understand why the lookup uses `false` and the delete has `isLocalNode`, I would expect `isLocalNode` for both. Could you please explain?

Reply:
You are right. I missed it. Good catch!